aiosmtpd/tests/test_smtp.py

```diff
@@ -202,1404 +211,1420 @@ class SleepingHeloHandler:
+        self._helo(client)
+        # Monkeypatching
+        smtpd: Server = exposing_controller.smtpd
+        spy: MagicMock = mocker.spy(smtpd._writer, "close")
+        # Do some stuff
+        client.docmd("MAIL FROM: <anne@example.com>")
+        client.docmd("RCPT TO: <bart@example.com>")
+        # Entering portion of code where hang is possible (upon assertion fail), so
+        # we must wrap with "try..finally". See pytest-dev/pytest#7989
+        try:
+            code, _ = client.docmd("DATA")
+            assert code == 354
             # Start sending the DATA but reset the connection before that
             # completes, i.e. before the .\r\n
-            client.send(b'From: <anne@example.com>')
+            client.send(b"From: <anne@example.com>")
             reset_connection(client)
-            # The connection should be disconnected, so trying to do another
-            # command from here will give us an exception.  In GH#62, the
-            # server just hung.
-            self.assertRaises(SMTPServerDisconnected, client.noop)
-
-    def test_connection_reset_during_command(self):
-        with SMTP(*self.address) as client:
-            client.helo('example.com')
-            # Start sending a command but reset the connection before that
-            # completes, i.e. before the \r\n
-            client.send('MAIL FROM: <anne')
-            reset_connection(client)
-            # The connection should be disconnected, so trying to do another
-            # command from here will give us an exception.  In GH#62, the
-            # server just hung.
-            self.assertRaises(SMTPServerDisconnected, client.noop)
-
-    def test_close_in_command(self):
-        with SMTP(*self.address) as client:
+            time.sleep(ASYNCIO_CATCHUP_DELAY)
+            # Apparently within that delay, ._writer.close() invoked several times
+            # That is okay; we just want to ensure that it's invoked at least once.
+            assert spy.call_count > 0
+        finally:
+            exposing_controller.stop()
+
+    def test_connreset_during_command(self, mocker, exposing_controller, client):
+        # Trigger factory() to produce the smtpd server
+        self._helo(client)
+        time.sleep(ASYNCIO_CATCHUP_DELAY)
+        smtpd: Server = exposing_controller.smtpd
+        spy: MagicMock = mocker.spy(smtpd._writer, "close")
+        # Start sending a command but reset the connection before that
+        # completes, i.e. before the \r\n
+        client.send("MAIL FROM: <anne")
+        reset_connection(client)
+        time.sleep(ASYNCIO_CATCHUP_DELAY)
+        # Should be called at least once. (In practice, almost certainly just once.)
+        assert spy.call_count > 0
+
+    def test_close_in_command(self, base_controller, client):
+        #
+        # What exactly are we testing in this test case, actually?
+        #
+        # Don't include the CRLF.
+        client.send("FOO")
+        client.close()
```

Comment:
Not really sure what this test case is for...?

I expanded the test case with an additional test case below.